src/os/bluestore/BlueFS.cc

```diff
@@ -3397,6 +3397,7 @@ int BlueFS::truncate(FileWriter *h, uint64_t offset)/*_WF_L*/
   std::lock_guard ll(log.lock);
   vselector->sub_usage(h->file->vselector_hint, h->file->fnode.size);
   h->file->fnode.size = offset;
+  h->file->is_dirty = true;
   vselector->add_usage(h->file->vselector_hint, h->file->fnode.size);
   log.t.op_file_update_inc(h->file->fnode);
```

Comment:
It unfortunate we have now both direct update of `log.t` and signaling `is_dirty`.
How about:
```
{
  std::lock_guard ll(log.lock);
  vselector->sub_usage(h->file->vselector_hint, h->file->fnode.size);
  h->file->fnode.size = offset;
  vselector->add_usage(h->file->vselector_hint, h->file->fnode.size);
  log.t.op_file_update_inc(h->file->fnode);
  }
  _flush_and_sync_log_LD(0);
  ```
  This way we would be in-sync right after truncate(), without need to fsync().

Reply:
Forget it.
I just realized, that sequence `truncate(h); fsync(h);` should give us proper results even if we ship `sync_metadata()` or `umount()`.